openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,236 @@
+package managementcpusoverride
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	PluginName                 = "autoscaling.openshift.io/ManagementCPUsOverride"
+	podWorkloadLabel           = "workload.openshift.io"
+	managedCapacityLabel       = "workload.openshift.io/cores"
+	containerCPUsAnnotation    = "io.openshift.workload.management.cpu"
+	containerCPUSetAnnotation  = "io.openshift.workload.management.cpuset"
+	namespaceAllowedAnnotation = "workload.openshift.io/allowed"
+	workloadTypeManagement     = "management"
+)
+
+var _ = initializer.WantsExternalKubeInformerFactory(&managementCPUsOverride{})
+var _ = admission.MutationInterface(&managementCPUsOverride{})
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(config io.Reader) (admission.Interface, error) {
+			return &managementCPUsOverride{
+				Handler: admission.NewHandler(admission.Create),
+			}, nil
+		})
+}
+
+type managementCPUsOverride struct {
+	*admission.Handler
+	nsLister corev1listers.NamespaceLister
+}
+
+func (a *managementCPUsOverride) SetExternalKubeInformerFactory(kubeInformers informers.SharedInformerFactory) {
+	a.nsLister = kubeInformers.Core().V1().Namespaces().Lister()
+}
+
+func (a *managementCPUsOverride) ValidateInitialization() error {
+	if a.nsLister == nil {
+		return fmt.Errorf("%s did not get a namespace lister", PluginName)
+	}
+	return nil
+}
+
+func (a *managementCPUsOverride) Admit(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	if attr.GetResource().GroupResource() != coreapi.Resource("pods") || attr.GetSubresource() != "" {
+		return nil
+	}
+
+	pod, ok := attr.GetObject().(*coreapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attr, fmt.Errorf("unexpected object: %#v", attr.GetObject()))
+	}
+
```

Comment:
skip mirror pods of static pods or don't they overlap with this logic here?

Reply:
it a good point, but if the kubelet deleted container CPUs requests from resources for static pods, the admission plugin will not do additional updates because of
```
if _, ok := c.Resources.Requests[coreapi.ResourceCPU]; !ok {
    continue
}
```

Reply:
May be worth adding a comment about it.

Reply:
done